robocop/checkers/naming.py

```diff
@@ -148,7 +148,7 @@ def check_keyword_naming(self, keyword_name, node):  # noqa
         if '_' in keyword_name:
             self.report("underscore-in-keyword-name", node=node)
         words = self.letter_pattern.sub(' ', keyword_name).split(' ')
-        if any(not (word.istitle() or word.isupper()) for word in words if word):
+        if any(word[0].islower() for word in words if word):
```

Comment:
Yes, it would pass and I think it should work that way because sometimes there can be something like:
`Check If RoboCop Works` and it shouldn't raise a warning. What do you think?

Reply:
Yes I agree. There could be typos in the middle of the name but honestly it's impossible to catch everything. That's why we should only focus on title casing (first letter upper).

Reply:
@mnojek Since we will simplify how this keyword work maybe we could simplify the code? Regex pattern will not be necessary. Proposed algo:

Replace _ with spaces, split on space, on every first char if it is letter (so ignore numbers, apostrophes, quotes etc) check if its upper